plugins/org.eclipse.xtext/src/org/eclipse/xtext/serializer/sequencer/ContextFinder.java

```diff
@@ -50,27 +53,25 @@
 	@Inject
 	protected IAssignmentFinder assignmentFinder;
 
-	protected List<IConstraintContext> constraintContexts;
-
-	protected Map<Pair<EObject, EClass>, IConstraint> constraints;
+	protected Map<IContext, IConstraint> constraints;
 
 	@Inject
-	protected RuleNames ruleNames;
+	protected IGrammarConstraintProvider grammarConstraintProvider;
 
 	@Inject
-	protected IGrammarConstraintProvider grammarConstraintProvider;
+	protected ISemanticNodeProvider nodesProvider;
 
 	@Inject
-	protected TransientValueUtil transientValueUtil;
+	protected RuleNames ruleNames;
 
 	@Inject
 	protected ITransientValueService transientValues;
 
 	@Inject
-	protected ISemanticNodeProvider nodesProvider;
+	protected TransientValueUtil transientValueUtil;
 
-	protected Iterable<AbstractElement> findAssignedElements(EObject obj, EStructuralFeature feature,
-			Iterable<AbstractElement> candidates) {
+	protected Set<AbstractElement> findAssignedElements(EObject obj, EStructuralFeature feature,
```

Comment:
Should this API require ordering guarantees, e.g. by using LinkedHashMultimap as input?